### PR TITLE
User: View and Edit Profile

### PR DESCRIPTION
**Pull request by samraarshad63**

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -1,3 +1,15 @@
...
+  def after_sign_in_path_for(current_user)
+    user_path(current_user)
+  end
+
+  def after_sign_up_path_for(current_user)
+    user_path(current_user)
+  end
+
+  def after_sign_out_path_for(current_user)
+   	users_path
```
Comment (faisalraza) [inline app/controllers/application_controller.rb]: Indentation issue

File: app/views/devise/registrations/edit.html.erb [hunk 2]
```diff
@@ -3,9 +3,29 @@
 <%= form_for(resource, as: resource_name, url: registration_path(resource_name), html: { method: :put }) do |f| %>
   <%= render "devise/shared/error_messages", resource: resource %>
 
+  <div class="field">
+    <%= f.label :firstname %><br />
+    <%= f.email_field :firstname, autofocus: true, autocomplete: "email" %>
```
Comment (faisalraza) [inline app/views/devise/registrations/edit.html.erb]: Do we need `autocomplete: "email"`?

File: app/views/devise/registrations/edit.html.erb [hunk 3]
```diff
@@ -3,9 +3,29 @@
 <%= form_for(resource, as: resource_name, url: registration_path(resource_name), html: { method: :put }) do |f| %>
   <%= render "devise/shared/error_messages", resource: resource %>
 
+  <div class="field">
+    <%= f.label :firstname %><br />
```
Comment (faisalraza) [inline app/views/devise/registrations/edit.html.erb]: Lets try to prevent `<br/>` tag

File: app/views/users/index.html.erb [hunk 4]
```diff
@@ -0,0 +1 @@
+<h1>Users#index</h1>
```
Comment (faisalraza) [inline app/views/users/index.html.erb]: Lets rename heading like `Users Listing`

File: app/views/users/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,16 @@
+<%= link_to 'MyProfile', user_path(current_user) %>
+<%= link_to 'Signout', destroy_user_session_path, method: :delete %>
+
+<h1>User profile</h1>
+<strong>Firstname: </strong><%= @user.firstname %>
```
Comment (faisalraza) [inline app/views/users/show.html.erb]: Can we write these kind of lines under `<div>` or `<p>` tag and remove `<br/>`?

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (samraarshad63) [reply app/controllers/application_controller.rb:13]: There is some problem with git! it sometimes shows different indentations from the original file in my editor. I'm looking into it.

File: app/views/devise/registrations/edit.html.erb (hunk 2 above)
Comment (samraarshad63) [reply app/views/devise/registrations/edit.html.erb:8]: we don't need it! I just added it for ease.

File: config/routes.rb [hunk 6]
```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
+  devise_for :users, controllers: { registrations: "users/registrations" }
   devise_for :admins
+
   get 'home/index'
-  devise_for :users, controllers: { registrations: "users/registrations" }
   get "/pages/:page" => "pages#show", as: :pages
+  
+  resources :users, :only =>[:index, :show]
```
Comment (faisalraza) [inline config/routes.rb]: Lets use colon notation ie, `only: [:show, :index]`

File: app/views/users/index.html.erb (hunk 4 above)
Comment (samraarshad63) [reply app/views/users/index.html.erb]: i still have to implement this task i-e the user listings task. I'll update it.

File: app/views/users/show.html.erb (hunk 5 above)
Comment (samraarshad63) [reply app/views/users/show.html.erb]: yes we can! I'll do that

File: config/routes.rb (hunk 6 above)
Comment (samraarshad63) [reply config/routes.rb]: okay!

File: app/views/devise/registrations/edit.html.erb (hunk 3 above)
Comment (samraarshad63) [reply app/views/devise/registrations/edit.html.erb]: okay! noted.

File: app/views/devise/registrations/edit.html.erb [hunk 7]
```diff
@@ -4,30 +4,49 @@
...
+  </div>
+
+  <div class="field">
+    <div><%= f.label :email %></div>
+    <%= f.email_field :email, autocomplete: "email" %>
+  </div>
+
+  <div class="field">
+    <div><%= f.label :country %></div>
+    <%= f.country_select :country, autocomplete: "country"  %>
```
Comment (faisalraza) [inline app/views/devise/registrations/edit.html.erb]: Extra space after `"country"`

File: app/views/users/show.html.erb [hunk 8]
```diff
@@ -0,0 +1,11 @@
+<%= link_to 'MyProfile', user_path(current_user) %>
+<%= link_to 'Signout', destroy_user_session_path, method: :delete %>
+
+<h1>User profile</h1>
+<div><strong>Firstname: </strong><%= @user.firstname %></div>
```
Comment (faisalraza) [inline app/views/users/show.html.erb]: Lets update indentation like below:
```
<div>
  <strong>Firstname: </strong>
  <%= @user.firstname %>
</div>
```

Comment (samraarshad63) [reply app/views/users/show.html.erb]: okay!

Comment (samraarshad63) [reply app/views/users/show.html.erb]: every div should also be separated by an empty line right?

**Review by faisalraza — APPROVED** [review]

**Pull request merged**